test/eni/IPRelease.go

```diff
@@ -0,0 +1,172 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package eniTest
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	. "github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/check.v1"
+
+	"github.com/cilium/cilium/pkg/logging"
+	. "github.com/cilium/cilium/test/ginkgo-ext"
+	"github.com/cilium/cilium/test/helpers"
+	"github.com/cilium/cilium/test/nodegroups"
+)
+
+func getAllocatedIPsOnNode(kubectl *helpers.Kubectl, nodeName string) (int, error) {
+	cn := kubectl.GetCiliumNode(nodeName)
+	if cn == nil {
+		return 0, fmt.Errorf("Unable to fetch cn %s", nodeName)
+	}
+	return len(cn.Spec.IPAM.Pool), nil
+}
+
+func hasRequiredIPsAllocated(kubectl *helpers.Kubectl, nodeName string, n int) func() bool {
+	return func() bool {
+		cn := kubectl.GetCiliumNode(nodeName)
+		if cn == nil {
+			return false
+		}
+		return len(cn.Spec.IPAM.Pool) == n
+	}
+}
+
+func hasRequiredIPBuffer(kubectl *helpers.Kubectl, nodeName string, n int) func() bool {
+	return func() bool {
+		cn := kubectl.GetCiliumNode(nodeName)
+		if cn == nil {
+			return false
+		}
+		return (len(cn.Spec.IPAM.Pool) - len(cn.Status.IPAM.Used)) == n
+	}
+}
+
+func hasNoIPsInHandshake(kubectl *helpers.Kubectl, nodeName string) func() bool {
+	return func() bool {
+		cn := kubectl.GetCiliumNode(nodeName)
+		if cn == nil {
+			return false
+		}
+		return len(cn.Status.IPAM.ReleaseIPs) == 0
+	}
+}
+
+var _ = Describe("EniIPReleaseHandshakeTest", func() {
```

Comment:
I think we could probably make this structure a bit cleaner by doing
```suggestion
var _ = Describe("ENI", func() {
```
and putting the `IPReleaseHandShakeTest` to be inside a `Context()`. This would mean moving all the `It()` into the `Context()` as well. This will allow us to create new `Context()` under the `ENI` group for other tests as well.